aptos-move/aptos-debugger/src/lib.rs

```diff
@@ -76,11 +80,70 @@ impl AptosDebugger {
             begin += epoch_result.len() as u64;
             limit -= epoch_result.len() as u64;
             txns = txns.split_off(epoch_result.len());
+            let epoch_txn_infos = txn_infos.drain(0..epoch_result.len()).collect::<Vec<_>>();
+            Self::print_mismatches(&epoch_result, &epoch_txn_infos, begin);
+
             ret.append(&mut epoch_result);
         }
         Ok(ret)
     }
 
+    fn print_mismatches(
+        txn_outputs: &[TransactionOutput],
+        expected_txn_infos: &[TransactionInfo],
+        first_version: Version,
+    ) {
+        for idx in 0..txn_outputs.len() {
+            let txn_output = &txn_outputs[idx];
+            let txn_info = &expected_txn_infos[idx];
+            let version = first_version + idx as Version;
+            let expected_txn_status: TransactionStatus = txn_info.status().clone().into();
+            if txn_output.status() != &expected_txn_status {
+                println!(
+                    "Mismatch: ver:{} status:{:?} on_chain:{:?}",
+                    version,
+                    txn_output.status(),
+                    expected_txn_status,
+                );
+            }
+
+            if txn_output.gas_used() != txn_info.gas_used() {
+                println!(
+                    "Mismatch: ver:{} gas_used:{} on_chain:{}",
+                    version,
+                    txn_output.gas_used(),
+                    txn_info.gas_used(),
+                );
+            }
+
+            let write_set_hash = txn_output.write_set().hash();
+            if write_set_hash != txn_info.state_change_hash() {
+                println!(
```

Comment:
Would it be better to print out the content of the writeset?

Reply:
The entire txn_output is printed out outside this call, no mater if there's a mismatch. (it's pretty messy when running in batch). I'd use the explorer anyway even for a single txn.